### PR TITLE
[SPARK-11256] Mark all Stage/ResultStage/ShuffleMapStage internal state as private.

### PR DESCRIPTION
**Pull request by rxin**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala [hunk 1]
```diff
@@ -535,10 +535,8 @@ class DAGScheduler(
     jobIdToActiveJob -= job.jobId
     activeJobs -= job
     job.finalStage match {
-      case r: ResultStage =>
-        r.resultOfJob = None
-      case m: ShuffleMapStage =>
-        m.mapStageJobs = m.mapStageJobs.filter(_ != job)
+      case r: ResultStage => r.removeActiveJob()
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala]: one thing i'm thinking is whether we should move "removeActiveJob" and "addActiveJob" to Stage, and have both ResultStage and ShuffleMapStage implement those.

File: core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala [hunk 2]
```diff
@@ -43,35 +43,53 @@ private[spark] class ShuffleMapStage(
     val shuffleDep: ShuffleDependency[_, _, _])
   extends Stage(id, rdd, numTasks, parents, firstJobId, callSite) {
 
+  private[this] var _mapStageJobs: List[ActiveJob] = Nil
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala]: cc @mateiz can you explain when we can have more than one active jobs for a stage? Does this only happen if we do adaptive query execution and bypass the existing DAGScheduler to run shuffle map stages?

**Comment by rxin** [comment]
cc @kayousterhout

File: core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala (hunk 2 above)
Comment (mateiz) [inline core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala]: This is if multiple threads submit the same stage with submitMapStage. You're not bypassing the DAGScheduler when you do adaptive execution, you're just telling it to run a map stage even if there's no result stage downstream. But it still does all its standard stuff to figure out previous stages to launch, track dependencies when recovering from failures, etc.

**Comment by rxin** [comment]
Going to merge this since this mostly just cleans up code.

**Pull request closed without merging**